### PR TITLE
Look through function conversion expressions when querying the called function decl of a deprecated-renamed API in order to emit a fix-it

### PR DESCRIPTION
**Pull request by artemcm**
Resolves rdar://113768664

### CODE REVIEW

**Comment by artemcm** [comment]
@swift-ci test

**Review by tshortli — APPROVED** [review]

File: test/attr/rename_function_convesion.swift [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// REQUIRES: concurrency
+// REQUIRES: objc_interop
```
Comment (tshortli) [inline test/attr/rename_function_convesion.swift:2]: Does this test require `objc_interop`?

File: test/attr/rename_function_convesion.swift
Comment (tshortli) [inline test/attr/rename_function_convesion.swift:1]: `rename_function_convesion.swift` -> `rename_function_conversion.swift`

**Comment by artemcm** [comment]
@swift-ci test

**Comment by artemcm** [comment]
@swift-ci test Windows platform

**Pull request merged**